server/src/main/java/io/pravega/schemaregistry/server/rest/resources/GroupResourceImpl.java

```diff
@@ -0,0 +1,690 @@
+                                                  ContinuationToken.EMPTY.toString() : result.getKey().toString();
+                                          groupsList.groups(
+                                                  result.getValue().stream().collect(
+                                                          Collectors.toMap(Map.Entry::getKey, x -> ModelHelper.encode(x.getValue()))))
+                                                    .setContinuationToken(contToken);
+                                      })
+                                   .thenApply(r -> Response.status(Status.OK).entity(groupsList).build())
+                                   .exceptionally(exception -> {
+                       log.warn("listGroups failed with exception: ", Exceptions.unwrap(exception));
+                       return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+                   })
+                                   .thenApply(response -> {
+                       asyncResponse.resume(response);
+                       return response;
+                   });
+        });
+    }
+
+    @Override
+    public void createGroup(CreateGroupRequest createGroupRequest, String namespace,
+                            AsyncResponse asyncResponse) {
+        String resource = Strings.isNullOrEmpty(namespace) ? getNamespaceResource() : getNamespaceResource(namespace);
+        withCompletion("createGroup", READ_UPDATE, resource, asyncResponse, () -> {
+            GroupProperties groupProperties = ModelHelper.decode(createGroupRequest.getGroupProperties());
+            String group = createGroupRequest.getGroupName();
+            return getRegistryService().createGroup(namespace, group, groupProperties)
+                                  .thenApply(createStatus -> {
+                                      if (!createStatus) {
+                                          log.info("group {} exists", group);
+                                          return Response.status(Status.CONFLICT).build();
+                                      }
+                                      log.info("group {} created", group);
+                                      return Response.status(Status.CREATED).build();
+                                  })
+                                  .exceptionally(exception -> {
+                                      log.warn("createGroup failed with exception: ", Exceptions.unwrap(exception));
+                                      return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+                                  });
+        }).thenApply(response -> {
+            asyncResponse.resume(response);
+            return response;
+        });
+    }
+
+    @Override
+    public void getGroupProperties(String group, String namespace,
+                                   AsyncResponse asyncResponse) {
+        String resource = Strings.isNullOrEmpty(namespace) ? getGroupResource(group) : 
```

Comment:
Input validation missing

Reply:
Group name is path param. Doesnt need validation.